src/commands/server/backup.ts

```diff
@@ -0,0 +1,292 @@
+export const backupRepositoryUrl = string({
+  char: 'r',
+  description: 'Full address of backup repository. Format is identical to restic.',
+  env: 'BACKUP_REPOSITORY_URL',
+  required: false,
+})
+
+export const BACKUP_REPOSITORY_PASSWORD_KEY = 'repository-password'
+export const backupRepositoryPassword = string({
+  char: 'p',
+  description: 'Password that is used to encrypt / decrypt backup repository content',
+  env: 'BACKUP_REPOSITORY_PASSWORD',
+  required: false,
+})
+
+export const BACKUP_REST_SERVER_USERNAME_KEY = 'username'
+export const backupRestServerUsername = string({
+  description: 'Username for authentication in backup REST server',
+  env: 'REST_SERVER_USERNAME',
+  required: false,
+})
+
+export const BACKUP_REST_SERVER_PASSWORD_KEY = 'password'
+export const backupRestServerPassword = string({
+  description: 'Authentication password for backup REST server',
+  env: 'REST_SERVER_PASSWORD',
+  required: false,
+})
+
+export const SSH_KEY_KEY = 'ssh-key'
+export const sshKey = string({
+  description: 'Private SSH key for authentication on SFTP server',
+  env: 'SSH_KEY',
+  required: false,
+})
+
+export const SSH_KEY_FILE_KEY = 'ssh-key-file'
+export const sshKeyFile = string({
+  description: 'PAth to file with private SSH key for authentication on SFTP server',
+  env: 'SSH_KEY_FILE',
+  required: false,
+  exclusive: [SSH_KEY_FILE_KEY],
+})
+
+export const AWS_ACCESS_KEY_ID_KEY = 'aws-access-key-id'
+export const awsAccessKeyId = string({
+  description: 'AWS access key ID',
+  env: 'AWS_ACCESS_KEY_ID',
+  required: false,
+})
+
+export const AWS_SECRET_ACCESS_KEY_KEY = 'aws-secret-access-key'
+export const awsSecretAccessKey = string({
+  description: 'AWS secret access key',
+  env: 'AWS_SECRET_ACCESS_KEY',
+  required: false,
+})
+
+export const BACKUP_SERVER_CONFIG_CR_NAME_KEY = 'backup-server-config-name'
+export const backupServerConfigName = string({
+  description: 'Name of custom resource with backup server config',
+  env: 'BACKUP_SERVER_CONFIG_NAME',
+  required: false,
+  exclusive: [BACKUP_REPOSITORY_URL_KEY, BACKUP_REPOSITORY_PASSWORD_KEY],
+})
+
+export default class Backup extends Command {
+  static description = 'Backup Eclipse Che installation'
+
+  static examples = [
+    '# Reuse existing backup configuration or create and use internal backup server if none exists:\n' +
+    'chectl server:backup',
+    '# Create and use configuration for REST backup server:\n' +
+    'chectl server:backup -r rest:http://my-sert-server.net:4000/che-backup -p repopassword',
+    '# Create and use configuration for AWS S3 (and API compatible) backup server (bucket should be precreated):\n' +
+    'chectl server:backup -r s3:s3.amazonaws.com/bucketche -p repopassword',
+    '# Create and use configuration for SFTP backup server:\n' +
+    'chectl server:backup -r sftp:user@my-server.net:/srv/sftp/che-data -p repopassword',
+  ]
+
+  static flags: flags.Input<any> = {
```

Comment:
I don't like an idea to provide password via cli. Can we found a better way?

Reply:
This is an example.
In general, it is possible to do:
```sh
chectl server:backup -r sftp:user@my-server.net:/srv/sftp/che-data -p $REPO_PASSWORD
```
or
```sh
export BACKUP_REPOSITORY_PASSWORD=password
chectl server:backup -r sftp:user@my-server.net:/srv/sftp/che-data
```

Do you want me to change the example to ` -p $REPO_PASSWORD` ?